modules/mod_xfer.c

```diff
@@ -2853,11 +2853,6 @@ MODRET xfer_retr(cmd_rec *cmd) {
 
   pr_alarms_unblock();
 
-  cnt_steps = session.xfer.file_size / 100;
-  if (cnt_steps == 0) {
-    cnt_steps = 1;
-  }
```

Comment:
Why remove this code?  It is there to deliberately prevent `mod_xfer` from performing high amounts of Scoreboard IO.  Is it to improve the accuracy of the Scoreboard-reported progress?

Reply:
PR_SCORE_XFER_DONE updates in xfer_stor were moved to pr_throttle_pause and got unified with PR_SCORE_XFER_LEN and PR_SCORE_XFER_ELAPSED updates, so any code responsible for updates in xfer_stor, including update throttling is no longer needed.

In general, I believe a time based approach for throttling scoreboard updates (e.g. update in every 5 seconds) is better than a percent based approach. So IMHO that should be implemented in the future for the updates in pr_throttle_pause instead of the current PR_TUNABLE_XFER_SCOREBOARD_UPDATES based approach.

I was wrong however in the 2nd point of my commit message (I misread the code), because of the throttling you mentioned XFER_DONE updates are not excessive indeed.